pandas-stubs/_typing.pyi

```diff
@@ -43,16 +41,14 @@ from pandas.core.dtypes.dtypes import ExtensionDtype
 
 ArrayLike = Union[ExtensionArray, np.ndarray]
 AnyArrayLike = Union[Index, Series, np.ndarray]
-PythonScalar = Union[str, int, float, bool, complex]
+PythonScalar = Union[str, bool, complex]
```

Comment:
PR to make these changes in pandas https://github.com/pandas-dev/pandas/pull/47850